app/AppKernel.php

```diff
@@ -91,23 +92,22 @@ private function loadReleaseFromComposer()
         }
 
         try {
-            list($version, $sha) = explode('@', \PackageVersions\Versions::getVersion('shopware/shopware'));
+            list($version, $sha) = explode('@', Versions::getVersion('shopware/shopware'));
 
-            /*
-             * Trim leading v from versions like "v5.4.6"
-             */
-            $version = ltrim(strtolower($version), 'v');
+            // Split versions like "v5.5.0-RC1" into "v5.5.0" and "RC1"
+            list($strippedVersion, $stability) = explode('-', $version);
```

Comment:
So much string manipulation going on here.
Would't it be easier to use the regex`s match to get the already well defined version number.
No case and prefix manipulations necessary.

Reply:
I'll fix it :)

Reply:
try #^v?(?P<version>\d+\.\d\.\d)(?:-(?P<version_text>[\d\w]+))$#
i like regex :)

Reply:
```
#^v?(?P<version>\d+\.\d\.\d)(?:-(?P<version_text>[\d\w]+))$#
                              :-(
```
Why is your regex so sad? 🙁